lib/adagrams.rb

```diff
@@ -0,0 +1,109 @@
+# wave1 code
+def draw_letters
+  pool_of_letters = {
+    "A" => 9, "B" => 2, "C" => 2, "D" => 4, "E" => 12, "F" => 2, "G" => 3, "H" => 2, "I" => 9, "J" => 1, "K" => 1, "L" => 4, "M" => 2, "N" => 6, "O" => 8, "P" => 2, "Q" => 1, "R" => 6, "S" => 4, "T" => 6, "U" => 4, "V" => 2, "W" =>2, "X" => 1, "Y" => 2, "Z" => 1
+  }
+
+  players_letters = []
+  alphabet = ("A".."Z").to_a
+  
+  until players_letters.length == 10
+    random_letter = alphabet.sample
+    if pool_of_letters[random_letter] && pool_of_letters[random_letter] > 0
+      players_letters << random_letter
+      pool_of_letters[random_letter] -= 1
+    end
+  end
+  return players_letters
+end
+
+# wave2 code
+def uses_available_letters?(input, letters_in_hand)
+  user_word = input.upcase.chars
+  word_hash = {}
+  user_word.each do |letter|
+    if word_hash["#{letter}"].nil?
+      word_hash["#{letter}"] = 1
+    else
+      word_hash["#{letter}"] += 1
+    end
+  end
+  
+  full_pile = {}
+  letters_in_hand.each do |letter|
+    if full_pile["#{letter}"].nil?
+      full_pile["#{letter}"] = 1
+    else
+      full_pile["#{letter}"] += 1
+    end
+  end
+  return word_hash.all? do |letter, count|
+    full_pile.has_key?(letter) && (full_pile[letter] >= word_hash[letter])
+  end
+end
+
+#Wave_3
+def score_word(word)
+  one_point_letter = %w[A E I O U L N R S T]
```

Comment:
While this if/elif control structure works, it means that the information about which letter has which score is locked into this piece of code, and can't easily be used elsewhere. For example, if you wanted to display the value of each letter in a hand, you would need to repeat this work.

An alternative approach would be to store the letter scores in a hash, something like this:

```ruby
LETTER_SCORES = {
  "A" => 1
  "B" => 3,
  "C" => 3,
  "D" => 2,
  # ...
}
```

Then to get the score for a letter, you can say `LETTER_SCORES[letter]`.